### PR TITLE
Separate kbld configs

### LINKED ISSUE #1961: Separate kbld configs
**Describe the feature request**
We would like to separate the kbld-config.yaml(s) into individual packages and remove the monolithic kbld-config.yaml.

This change allows a package to be built without having to fill in values for other images that aren't used by whatever package is being built i.e it decouples package/image building from each other w.r.t kbld'ing.

This requires some tweaks to the existing build tooling to use the new kbld-config in the individual package folders.

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [X] Test and Release
- [ ] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by christianang**
### What this PR does / why we need it

This PR separates the `kbld-config.yaml(s)` into individual packages and removes the monolithic `kbld-config.yaml`. 

This change is desirable because it allows a package to be built without having to fill in values for other images that aren't used by whatever package is being built i.e it decouples package/image building from each other w.r.t kbld'ing.

### Which issue(s) this PR fixes

Fixes #1961

### Describe testing done for PR

```
make build-cli-with-local-discovery
make local-registry
OCI_REGISTRY=localhost:5001 make docker-build
OCI_REGISTRY=localhost:5001 make docker-publish
OCI_REGISTRY=localhost:5001 make kbld-image-replace
PACKAGE_REPOSITORY=management make package-bundles
PACKAGE_REPOSITORY=standalone make package-bundles
```
We were then able to verify the contents of the built package bundles.

### Release note

```release-note

```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### Additional information

This also changed the `tkr-resolver` and `tkr-conversion` Makefile's `kbld-image-replace` targets no-op because they do not currently have any packages that use their docker images, so there is nothing to replace.

### CODE REVIEW

File: packages/kbld-config.yaml [hunk 1]
```diff
@@ -1,72 +0,0 @@
-apiVersion: kbld.k14s.io/v1alpha1
```
Comment (yharish991) [inline packages/kbld-config.yaml:1]: 💯

File: packages/management/addons-manager-v2/kbld-config.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apiVersion: kbld.k14s.io/v1alpha1
+kind: Config
+overrides:
+- image: addons-controller:latest
+  newImage: ""
+
```
Comment (yharish991) [inline packages/management/addons-manager-v2/kbld-config.yaml:6]: this empty line can be removed

File: packages/management/core-management-plugins/kbld-config.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+- image: tanzu-cli-plugins/feature-darwin-amd64:latest
+  newImage: ""
+#! tanzu-cliplugins windows-amd64
+- image: tanzu-cli-plugins/cluster-windows-amd64:latest
+  newImage: ""
+- image: tanzu-cli-plugins/kubernetes-release-windows-amd64:latest
+  newImage: ""
+- image: tanzu-cli-plugins/feature-windows-amd64:latest
+  newImage: ""
+
```
Comment (yharish991) [inline packages/management/core-management-plugins/kbld-config.yaml:25]: this empty line can be removed

File: packages/management/tkr-aws-machine-webhook/kbld-config.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+apiVersion: kbld.k14s.io/v1alpha1
+kind: Config
+overrides:
+- image: tkr-infra-machine-webhook:latest
+  newImage: ""
+
```
Comment (yharish991) [inline packages/management/tkr-aws-machine-webhook/kbld-config.yaml:6]: ditto..this empty line can be removed

File: packages/management/tkr-azure-machine-webhook/kbld-config.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+apiVersion: kbld.k14s.io/v1alpha1
+kind: Config
+overrides:
+- image: tkr-infra-machine-webhook:latest
+  newImage: ""
+
```
Comment (yharish991) [inline packages/management/tkr-azure-machine-webhook/kbld-config.yaml:6]: ditto..this empty line can be removed

File: packages/management/tkr-docker-machine-webhook/kbld-config.yaml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+apiVersion: kbld.k14s.io/v1alpha1
+kind: Config
+overrides:
+- image: tkr-infra-machine-webhook:latest
+  newImage: ""
+
```
Comment (yharish991) [inline packages/management/tkr-docker-machine-webhook/kbld-config.yaml:6]: ditto..this empty line can be removed

File: packages/management/tkr-vsphere-machine-webhook/kbld-config.yaml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+apiVersion: kbld.k14s.io/v1alpha1
+kind: Config
+overrides:
+- image: tkr-infra-machine-webhook:latest
+  newImage: ""
+
```
Comment (yharish991) [inline packages/management/tkr-vsphere-machine-webhook/kbld-config.yaml:6]: ditto..this empty line can be removed

File: pkg/v2/tkr/webhook/tkr-conversion/Makefile [hunk 8]
```diff
@@ -43,7 +43,7 @@ docker-publish: ## Publish docker image
 
 .PHONY: kbld-image-replace
 kbld-image-replace: ## Add newImage in kbld-config.yaml
-	cd ../../../../../hack/packages/kbld-image-replace && $(MAKE) run IMAGE=$(IMG_DEFAULT_NAME_TAG) NEW_IMAGE=$(IMG)
+	echo "No tkr-conversion package to kbld-image-replace"
```
Comment (yharish991) [inline pkg/v2/tkr/webhook/tkr-conversion/Makefile:46]:
```suggestion
	echo "No image in tkr-conversion package to kbld-image-replace"
```

File: pkg/v2/tkr/webhook/cluster/tkr-resolver/Makefile [hunk 9]
```diff
@@ -43,7 +43,7 @@ docker-publish: ## Publish docker image
 
 .PHONY: kbld-image-replace
 kbld-image-replace: ## Add newImage in kbld-config.yaml
-	cd ../../../../../../hack/packages/kbld-image-replace && $(MAKE) run IMAGE=$(IMG_DEFAULT_NAME_TAG) NEW_IMAGE=$(IMG)
+	echo "No tkr-resolver package to kbld-image-replace"
```
Comment (yharish991) [inline pkg/v2/tkr/webhook/cluster/tkr-resolver/Makefile:46]:
```suggestion
	echo "No image in tkr-resolver package to kbld-image-replace"
```

File: pkg/v2/tkr/webhook/infra-machine/Makefile [hunk 10]
```diff
@@ -43,7 +43,11 @@ docker-publish: ## Publish docker image
 
 .PHONY: kbld-image-replace
 kbld-image-replace: ## Add newImage in kbld-config.yaml
-	cd ../../../../../hack/packages/kbld-image-replace && $(MAKE) run IMAGE=$(IMG_DEFAULT_NAME_TAG) NEW_IMAGE=$(IMG)
+	cd ../../../../../hack/packages/kbld-image-replace && \
+		go run main.go -kbld-config ../../../packages/management/tkr-aws-machine-webhook/kbld-config.yaml $(IMG_DEFAULT_NAME_TAG) $(IMG) && \
```
Comment (yharish991) [inline pkg/v2/tkr/webhook/infra-machine/Makefile:47]: can we cache the binary and reuse it, instead of building it each time?

**Review by yharish991 — CHANGES_REQUESTED** [review]

**Comment by yharish991** [comment]
Can you also file an issue for this pr and update the testing done? thanks :)

File: pkg/v2/tkr/webhook/infra-machine/Makefile (hunk 10 above)
Comment (adobley) [reply pkg/v2/tkr/webhook/infra-machine/Makefile:47]: I'm unsure what you mean by caching the binary. We could generate a kbld-config once and copy it to each `tkr-*-machine-webhook` folder. Is that what you had in mind?

Comment (flawedmatrix) [reply pkg/v2/tkr/webhook/infra-machine/Makefile:47]: Per our discussion I think we'll optimize/refactor this in another PR.

**Review by yharish991 — APPROVED** [review]
LGTM!! Thanks..

**Comment by adobley** [comment]
Howdy @vmware-tanzu/tanzu-framework-reviewers & @vmware-tanzu/tkg-addons-owners, when y'all have a moment we'd appreciate a review so we can get this merged.

Thanks!

File: addons/Makefile [hunk 11]
```diff
@@ -118,7 +118,8 @@ docker-build-and-publish: docker-build docker-publish kbld-image-replace
 
 .PHONY: kbld-image-replace
 kbld-image-replace: ## Add newImage in kbld-config.yaml
-	cd ../hack/packages/kbld-image-replace && $(MAKE) run IMAGE=$(IMG_DEFAULT_NAME_TAG) NEW_IMAGE=$(IMG)
+	cd ../hack/packages/kbld-image-replace && \
+	  go run main.go -kbld-config ../../../packages/management/addons-manager-v2/kbld-config.yaml $(IMG_DEFAULT_NAME_TAG) $(IMG)
```
Comment (maralavi) [inline addons/Makefile:122]: @shyaamsn I believe you've changed ```addons-manager-v2``` to ```addons-manager``` in [this PR](https://github.com/vmware-tanzu/tanzu-framework/pull/2315). Should the path ```../../../packages/management/addons-manager-v2/kbld-config.yaml```  here get changed to ```../../../packages/management/addons-manager/kbld-config.yaml``` ?

File: packages/management/addons-manager-v2/kbld-config.yaml [hunk 12]
```diff
@@ -0,0 +1,5 @@
+apiVersion: kbld.k14s.io/v1alpha1
```
Comment (maralavi) [inline packages/management/addons-manager-v2/kbld-config.yaml:1]: Same comment here as above. ```addons-manager-v2``` has changed to ```addons-manager```in [this PR](https://github.com/vmware-tanzu/tanzu-framework/pull/2315) . Please make appropriate changes. cc: @shyaamsn

File: addons/Makefile (hunk 11 above)
Comment (christianang) [reply addons/Makefile]: Thanks for the info. Updated.

File: packages/management/addons-manager-v2/kbld-config.yaml (hunk 12 above)
Comment (christianang) [reply packages/management/addons-manager-v2/kbld-config.yaml]: Updated.

**Review by maralavi — APPROVED** [review]

**Review by sabbey37 — APPROVED** [review]
Looks good, thank you for doing this!

**Review by benjaminapetersen — APPROVED** [review]
spiffy.

**Review by rajathagasthya — APPROVED** [review]

**Pull request merged**